frame/staking/src/benchmarking.rs

```diff
@@ -634,6 +644,81 @@ benchmarks! {
 	verify {
 		assert!(!Validators::<T>::contains_key(controller));
 	}
+
+	rebag {
+		// The most expensive case for this call:
+		//
+		// - It doesn't matter where in the origin bag the stash lies; the number of reads and
+		//   writes is constant. We can use the case that the stash is the only one in the origin
+		//   bag, for simplicity.
+		// - The destination bag is not empty, because then we need to update the `next` pointer
+		//   of the previous node in addition to the work we do otherwise.
+
+		use crate::voter_bags::{Bag, Node};
+
+		let make_validator = |n: u32, balance_factor: u32| -> Result<(T::AccountId, T::AccountId), &'static str> {
+			let (stash, controller) = create_stash_controller::<T>(n, balance_factor, Default::default())?;
+			whitelist_account!(controller);
+
+			let prefs = ValidatorPrefs::default();
+			// bond the full value of the stash
+			let free_balance = T::Currency::free_balance(&stash);
+			Staking::<T>::bond_extra(RawOrigin::Signed(stash.clone()).into(), free_balance)?;
+			Staking::<T>::validate(RawOrigin::Signed(controller.clone()).into(), prefs)?;
+
+			Ok((stash, controller))
+		};
+
+		// Clean up any existing state.
+		clear_validators_and_nominators::<T>();
+
+
+		// stash controls the node account
```

Comment:
what's the "node account"?

Reply:
Yeah, that's confusing. What I'm trying to get at is there are two voters in play here, one of which we care about / edit, and one which is just there to increase the complexity by already being present in the destination bag. In general I name the variables `thing` for things of the first type, and `other_thing` for things of the second. I'll think about how to clarify this.

Reply:
Hopefully https://github.com/paritytech/substrate/pull/9081/commits/22bc615a25c1f76ac42c1060061b74f9ab644fda expresses this all a bit more clearly.